src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        {
+            // this should be a soft reset
+            WriteRegister(Register.PWR_MGMT_1, 0x01);
+            Thread.Sleep(200);
+            _wakeOnMotion = false;
+        }
+
+        /// <summary>
+        /// Return true if the version of MPU9250 is the correct one
+        /// </summary>
+        /// <returns></returns>
+        public bool CheckVersion()
+        {
+            // Check if the version is thee correct one
+            return ReadByte(Register.WHO_AM_I) == 0x71;
+        }
+
+        /// <summary>
+        /// Get or set the sample diver mode
+        /// </summary>
+        public byte SampleRateDivider
+        {
+            get { return ReadByte(Register.SMPLRT_DIV); }
+            set { WriteRegister(Register.SMPLRT_DIV, value); }
+        }
+
+        /// <summary>
+        /// Get or set the elements to disable. 
+        /// It can be any axes of the accelerometer and or the gyroscope
+        /// </summary>
+        public DisableModes DisableModes
+        {
+            get { return (DisableModes)ReadByte(Register.PWR_MGMT_2); }
+            set { WriteRegister(Register.PWR_MGMT_2, (byte)value); }
+        }
+
+        /// <summary>
+        /// Cleanup everything
+        /// </summary>
+        public void Dispose()
+        {
+            if (_autoDispose)
+            {
+                _i2cDevice?.Dispose();
+                _i2cDevice = null;
+            }
+        }
+
+        #endregion
+
+        #region FIFO
+
+        /// <summary>
+        /// Get the number of elements to read from the FIFO
+        /// </summary>
+        public uint FifoCount
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2] { 0, 0 };
+                ReadByteArray(Register.FIFO_COUNTH, rawData);
+                return BinaryPrimitives.ReadUInt16BigEndian(rawData);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the FIFO modes
+        /// </summary>
+        public FifoModes FifoModes
+        {
+            get { return (FifoModes)(ReadByte(Register.FIFO_EN)); }
+            set
+            {
+                if (value != FifoModes.None)
+                {
+                    // Make sure the FIFO is enbalbe
+                    var usrCtl = (UserControls)ReadByte(Register.USER_CTRL);
+                    usrCtl = usrCtl | UserControls.FIFO_RST;
+                    WriteRegister(Register.USER_CTRL, (byte)usrCtl);
+                }
+                else
+                {
+                    // Deactivate FIFO
+                    var usrCtl = (UserControls)ReadByte(Register.USER_CTRL);
+                    usrCtl = usrCtl & ~UserControls.FIFO_RST;
+                    WriteRegister(Register.USER_CTRL, (byte)usrCtl);
+                }
+                WriteRegister(Register.FIFO_EN, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Read data in the FIFO, read as many data as the size of readData byte span
```

Comment:
Do you mind adding in parenthesis what FIFO stands for?

Also, should this return a struct with the result given that it seems to return a bunch of different accelerometer and giroscope data among other things? So for example a struct that has 2 vectors for acceleration and gyroscope data, plus the temperature and the sensor data. That way this function becomes more useful than just something that returns bytes.

Reply:
Added the definition. Depending on what you select to read, the array will have different size. As explained in the summary, the data are coming in a specific order.
If you are using the Fifo, you know what you want to read, so you know the size of the buffer and the order of the data.
Also, if you are using Fifo, it's most likely because you want speed and a very short loop, so adding conversion to a struct, copying data is actually taking processing time that most likely is not needed otherwise, you won't use the Fifo. So I can add a nice function on top returning a struct but I'll keep as well this one for performance reasons.